src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/generate_static_shapes.hpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/preprocess/color_format.hpp"
+
+#include "shared_test_classes/base/utils/ranges.hpp"
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+
+namespace ov {
+namespace test {
+namespace utils {
+
+using ShapesMap = std::map<ov::NodeTypeInfo, std::function<InputShape(
```

Comment:
What do you think about to replace used args (`const std::shared_ptr<ov::Node>& node, const std::shared_ptr<ov::op::v0::Parameter>& param`) to `const std::shared_ptr<ov::Node>& node, size_t in_port_id`?

Looks like it will help to avoid some problems to generate some problems based on `in_port_id`? Because we can get parameter (`const auto param = node->get_input_node_shared_ptr(in_port_id)` explicitly and generate input shape based on port_id (because `port_id` can be important)...